src/engine/source/kvdb/include/kvdb/kvdbManager.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef _KVDBMANAGER_H
+#define _KVDBMANAGER_H
+
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include "kvdb.hpp"
+
+class KVDBManager
+{
+    KVDBManager();
+    ~KVDBManager();
+    const std::string FOLDER = "/var/ossec/queue/db/kvdb/";
+    using DBMap = std::unordered_map<std::string, KVDB *>;
+    DBMap available_kvdbs;
+    bool addDB(KVDB *DB);
+    KVDB invalidDB = KVDB(); // TODO: Make this static?
+
+public:
+    static KVDBManager &getInstance()
+    {
+        static KVDBManager instance;
+        return instance;
+    }
+    KVDBManager(KVDBManager const &) = delete;
+    void operator=(KVDBManager const &) = delete;
+    bool createDB(const std::string &Name, bool replace = true);
```

Comment:
`createDB` should return the created db or we are going to have the pattern below everywhere
```
KVDBManager::getInstance().createDb("name");
KVDBManager::getInstance().getDb("name");
```

Reply:
I have change the return value in order to be able to use
`KVDB &newKvdb = kvdbManager.createDB("NEW_DB");`
But theres also the option of using it as
```
    kvdbManager.deleteDB("NEW_DB");
    KVDB &deletedKvdb = kvdbManager.getDB("NEW_DB");
```